tests/foreman/cli/test_hostgroup.py

```diff
@@ -554,19 +568,31 @@ def test_positive_create_with_multiple_entities_ids(self):
         self.assertIn(loc['id'], hostgroup['locations'][0]['id'])
         self.assertEqual(
             env['id'], hostgroup['puppet-environment']['environment_id'])
-        self.assertEqual(proxy['id'], hostgroup['puppet-master-proxy-id'])
-        self.assertEqual(proxy['id'], hostgroup['puppet-ca-proxy-id'])
-        self.assertEqual(domain['id'], hostgroup['domain']['domain_id'])
+        self.assertEqual(
+            proxy['id'],
+            hostgroup['puppet-master-proxy']['puppet_proxy_id']
+        )
+        self.assertEqual(
+            proxy['id'],
+            hostgroup['puppet-master-proxy']['puppet_ca_proxy_id']
+        )
+        self.assertEqual(
+            domain['id'],
+            hostgroup['network']['domain']['domain_id']
+        )
         self.assertEqual(
                 subnet['id'],
-                hostgroup['network']['subnet-ipv4']['id'])
+                hostgroup['network']['domain']['subnet_id'])
         self.assertEqual(
-            arch['id'], hostgroup['architecture']['architecture_id'])
+            arch['id'], hostgroup['network']['domain']['architecture_id'])
         self.assertEqual(
-            ptable['id'], hostgroup['partition-table']['ptable_id'])
-        self.assertEqual(media['id'], hostgroup['medium']['medium_id'])
+            ptable['id'], hostgroup['network']['domain']['ptable_id'])
+        self.assertEqual(
+            media['id'],
+            hostgroup['network']['domain']['medium_id']
+        )
         self.assertEqual(
-            os['id'], hostgroup['operating-system']['operatingsystem_id'])
+            os['id'], hostgroup['network']['domain']['operatingsystem_id'])
```

Comment:
Why we you switched from key operating-system to retrieve  Os is
portion from json returned data
```
{
  "Id": 1,
  "Name": "mhQhb9",
  "Title": "mhQhb9",
  "Operating System": {
    "content_source_id": 1,
    "content_source_name": "ibm-ls22-05.rhts.eng.brq.redhat.com",
    "content_view_id": 5,
    "content_view_name": "NHooKRwrMU",
    "lifecycle_environment_id": 4,
    "lifecycle_environment_name": "pO8BT4",
    "kickstart_repository_id": null,
    "kickstart_repository_name": null,
    "subnet_id": 2,
    "subnet_name": "BhYE1xBV",
    "operatingsystem_id": 2,
    "operatingsystem_name": "3T3TUR 1.8",
    "domain_id": 2,
    "domain_name": "dzwmuu5ek6",
    "environment_id": 3,
    "environment_name": "XyQLDm",
    "compute_profile_id": null,
    "compute_profile_name": null,
    "ancestry": null,
    "parent_id": null,
    "parent_name": null,
    "ptable_id": 137,
    "ptable_name": "y7EY9Caqf9",
    "medium_id": 10,
    "medium_name": "jlV7Mq",
    "pxe_loader": null,
    "subnet6_id": null,
    "subnet6_name": null,
    "architecture_id": 3,
    "architecture_name": "vw2fMo98tJ",
    "realm_id": null,
...
```
same for architecture and ptable think they are more related to os than to network, what do you think?